scripts/core/CONTRIB.rst

```diff
@@ -1,71 +1,177 @@
+
+    *   The design phase of the feature is expected to span multiple oneAPI
+        releases.
+
+    *   A proof of concept implementation exists for a single adapter but
+        multiple adapters are intended to be supported.
 
-Updating the Specification
-==========================
+    If the feature in question matches any of these criteria, please refer to
+    the `Experimental Features`_ section, otherwise refer to the `Core
+    Features`_ section. If you are unsure how to proceed please `create an
+    issue <https://github.com/oneapi-src/unified-runtime/issues/new>`_ asking
+    for clarification.
 
-The specification and many other components of Unified Runtime are generated 
-from a set of ``*.yml`` files located in the ``scripts/core/`` directory of the 
-project. To make a change to the specification you should modify only ``*.yml`` 
-files in this directory. To see the changes reflected in the specification you 
-`must` build the ``generate`` target. When making changes to the specification 
-you should commit all changes generated by the script as part of your change. 
-Refer to ``scripts/YaML.md`` on the yaml syntax as well as detailed descriptions 
-of required fields for each type of entity.
+Generating Source
+=================
 
-Naming Conventions
-~~~~~~~~~~~~~~~~~~
+The specification and many other components in the Unified Runtime repository
+are generated from a set of YAML_ files which are used as inputs to a Mako_
+based templating system. The YAML_ file syntax is defined `YAML syntax`_. To
+generate the outputs of the Mako_ templates a build directory must be
+configured, instructions are available in the `README
+<https://github.com/oneapi-src/unified-runtime/blob/main/README.md>`_ file.
+Upon successfully configuring a build directory, generate the outputs with the
+following command (or suitable build system equivalent):
 
-Additional to what is described in the Yaml syntax document. You should follow
-the following naming convention for argument names.
+.. code-block:: console
 
-    * Entry-Points use ``camelCase`` conventions i.e ``${x}DeviceGet``.
+    $ cmake --build build --target generate
 
-    * Argument names are also `camelCase` and are prefixed with a type 
-      indicator:
+.. _YAML: https://yaml.org/
+.. _Mako: https://www.makotemplates.org/
+.. _YAML syntax:
+   https://github.com/oneapi-src/unified-runtime/blob/main/scripts/YaML.md
 
-        * Pointer arguments are prefixed with ``p`` or ``pp`` for double pointer 
-          i.e. ``pEvent``.
+Writing YAML
+============
 
-        * Handle arguments are prefixed with ``h`` i.e. ``hQueue``.
+Please read the :ref:`core/INTRO:Naming Convention` section prior to making a
+contribuition and refer to the `YAML syntax`_ for specifics of how to define
+the required contructs.
 
+When writing ``*.yml`` files and ``ur`` or ``UR`` should exist in the output
+use ``$${'x'}`` or ``$${'X'}`` respectively. These will be replaced while
```

Comment:
I believe it's not rendered well. I can see on your preview:
```
$X}
```

Reply:
That's odd, I see this
![image](https://github.com/oneapi-src/unified-runtime/assets/1625763/2b5c5d33-a692-4588-86cc-d3ca7ea0c13f)

Reply:
In both Firefox and Safari on macOS, which browser are you using @lukaszstolarczuk ?

Reply:
And in Chrome on Windows ![image](https://github.com/oneapi-src/unified-runtime/assets/1625763/b9b2bd94-9501-40ad-ba9f-edcc02b260f0)

Reply:
hmm.. I'm not sure what happened (something changed?), but now it looks ok - please disregard